### ISSUE TITLE
Why BATCH_PER_DEVICE set to 4 in finetune.sh?

### ISSUE THREAD

**Issue by yiyexy** [2U1/Qwen2-VL-Finetune#163]
Just as question about title.
The Qwen2VL do not have batch dim in forward, why can you set it to 4?
I'm follow your setting, the loss=0 and grad_norm=nan. 
This bug fixed after I change BATCH_PER_DEVICE to 1.

**Comment by 2U1** [comment]
```
# Sample messages for batch inference
messages1 = [
    {
        "role": "user",
        "content": [
            {"type": "image", "image": "file:///path/to/image1.jpg"},
            {"type": "image", "image": "file:///path/to/image2.jpg"},
            {"type": "text", "text": "What are the common elements in these pictures?"},
        ],
    }
]
messages2 = [
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": "Who are you?"},
]
# Combine messages for batch processing
messages = [messages1, messages2]

# Preparation for batch inference
texts = [
    processor.apply_chat_template(msg, tokenize=False, add_generation_prompt=True)
    for msg in messages
]
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=texts,
    images=image_inputs,
    videos=video_inputs,
    padding=True,
    return_tensors="pt",
)
inputs = inputs.to("cuda")

# Batch Inference
generated_ids = model.generate(**inputs, max_new_tokens=128)
generated_ids_trimmed = [
    out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
]
output_texts = processor.batch_decode(
    generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=False
)
print(output_texts)
```

This batch inference example is from the official repo. You could batch the input to the model.
There are no batch size in the `pixel_values`, but they are concatenated in the sequence dimension so its not a problem.

**Comment by 2U1** [comment]
Also, I think the problem is from Flash-Attention 2 (FA2). 
The “loss = 0 / grad_norm = nan” symptoms appear when you install a Windows (or old Linux) FA2 wheel that only contains the forward kernels—training breaks the moment the code path reaches those kernels, which happens as soon as the per‑device batch is > 1. Certain pre‑built wheels (especially on Windows) omit the backward pass. The FA2 docs even annotate some helper kernels with “Note: Does not support backward pass.”
